### PR TITLE
Update Terraform CLI versions to latest in test workflow

### PR DESCRIPTION
**Pull request by aareet**
TF 0.14 updated from 0.14.5 to 0.14.7

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -53,7 +53,8 @@ jobs:
         terraform:
           - '0.12.30'
           - '0.13.6'
```
Comment (bflad) [inline .github/workflows/test.yml:55]:
```suggestion
          - '0.12.31'
          - '0.13.7'
```

Comment (bflad) [inline .github/workflows/test.yml:55]:
```suggestion
          - '0.12.31'
          - '0.13.7'
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -53,7 +53,8 @@ jobs:
...
-          - '0.14.5'
+          - '0.14.10'
+          - '0.15.0-rc2'
```
Comment (bflad) [inline .github/workflows/test.yml:57]:
```suggestion
          - '0.14.11'
          - '0.15.1'
```

Comment (bflad) [inline .github/workflows/test.yml:57]:
```suggestion
          - '0.14.11'
          - '0.15.1'
```

Comment (bflad) [inline .github/workflows/test.yml:57]:
```suggestion
          - '0.14.11'
          - '0.15.1'
```

**Review by bflad — CHANGES_REQUESTED** [review]
Could you please update these to all the latest versions please? I think we will be looking at ways to specify only major.minor versions to reduce maintenance of these. 👍

**Comment by aareet** [comment]
@bflad @jrhouston has done some work in the helm provider to dynamically fetch the latest version of Terraform for the tests

**Comment by jrhouston** [comment]
> @bflad @jrhouston has done some work in the helm provider to dynamically fetch the latest version of Terraform for the tests

I have this: https://github.com/hashicorp/terraform-provider-helm/blob/main/.github/workflows/tests.yaml#L11 although I want to turn it into a GitHub action of it's own if we're going to share it so we can update it to use the releases API whenever it lands. The script is kind of brittle right now.

**Comment by bflad** [comment]
Nice yeah, we were looking at setup-terraform since it can accept version constraints (e.g. passed in via matrix), but longer term its likely that hcinstall might natively get the functionality as well.

File: .github/workflows/test.yml (hunk 1 above)
Comment (bflad) [reply .github/workflows/test.yml:55]: Not sure why GitHub opened multiple comments, but going to also include this for consistency. 👍

**Comment by bflad** [comment]
Guessing the failures here are because we need to pull in https://github.com/hashicorp/terraform-provider-external/pull/69 -- will work on getting that in now.

**Review by bflad — APPROVED** [review]

**Pull request merged**